catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CommandSupport.java

```diff
@@ -78,41 +67,36 @@ protected void printSuccessMessage(String message) {
         printColor(SUCCESS_COLOR, message);
     }
 
+    /**
+     * Logic mimics {@link org.apache.karaf.main.StartupListener#showProgressBar}
+     *
+     * @param start         time started processing records
+     * @param totalCount    count of total records
+     * @param currentCount  count of records completed
+     * @return
+     */
     protected void printProgressAndFlush(long start, long totalCount, long currentCount) {
-        console.print(getProgressBar(currentCount, totalCount, start, System.currentTimeMillis()));
-        console.flush();
-    }
-
-    private String getProgressBar(long currentCount, long totalPossible, long start, long end) {
-
-        int notches = calculateNotches(currentCount, totalPossible);
-
-        int progressPercentage = calculateProgressPercentage(currentCount, totalPossible);
-
-        int rate = calculateRecordsPerSecond(currentCount, start, end);
-
-        String progressArrow = ">";
+        final int progressPercentage;
+        if (totalCount > 0) {
```

Comment:
Any concern with this getting called where currentCount > totalCount

Reply:
This method has never done any error checking.